docs/source/reference/deprecation.rst

```diff
@@ -397,37 +288,26 @@ device function, then a device function should be compiled.
 
 The first two cases cannot be differentiated in the absence of the ``device``
 kwarg - without it, it will not be clear from a signature alone whether a device
-function or global kernel should be compiled. In order to resolve this, support
-for eager compilation of device functions will be removed. Eager compilation
-with the ``@cuda.jit`` decorator will in future always imply the immediate
-compilation of a global kernel.
-
-Recommendations
----------------
+function or global kernel should be compiled. In order to resolve this, device
+functions will no longer be eagerly compiled. When a signature is provided to a
+device function, it will only be used to enforce the types of arguments that
+the function accepts.
 
-Any eagerly-compiled device functions should have their signature removed, e.g.:
+.. note::
 
-.. code-block:: python
-
-   @cuda.jit('int32(int32, int32)', device=True)
-   def f(x, y):
-       return x + y
-
-becomes:
-
-
-.. code-block:: python
+   In previous releases this notice stated that support for providing
+   signatures to device functions would be removed completely - however, this
+   precludes the common use case of enforcing the types that can be passed to a
+   device function (and the automatic insertion of casts that it implies) so
+   this notice has been updated to retain support for passing signatures.
 
-   @cuda.jit(device=True)
-   def f(x, y):
-       return x + y
 
 Schedule
 --------
 
 - In Numba 0.54: Eager compilation of device functions will be deprecated.
 - In Numba 0.55: Eager compilation of device functions will be unsupported and
-  attempts to eagerly compile device functions will raise an error.
+  the provision of signatures for device functions will only enforce casting.
 
 
 .. _rocm_unmaintained:
```

Comment:
Should this section also be removed, two releases have passed without comment?

Reply:
Sure, I can remove this.